src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -36,30 +37,134 @@ namespace winrt::TerminalApp::implementation
+            {
+                args->_TrimWhitespace = trimWhitespace.asBool();
+            }
+            return *args;
+        }
     };
 
-    struct NewTabWithProfileArgs : public NewTabWithProfileArgsT<NewTabWithProfileArgs>
+    struct NewTabArgs : public NewTabArgsT<NewTabArgs>
     {
-        NewTabWithProfileArgs() = default;
-        GETSET_PROPERTY(int32_t, ProfileIndex, 0);
+        NewTabArgs() = default;
+        GETSET_PROPERTY(Windows::Foundation::IReference<int32_t>, ProfileIndex, nullptr);
+
+        static constexpr std::string_view ProfileIndexKey{ "index" };
+
+    public:
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // IMPORTANT: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<NewTabArgs>();
+            if (auto profileIndex{ json[JsonKey(ProfileIndexKey)] })
+            {
+                args->_ProfileIndex = profileIndex.asInt();
+            }
+            return *args;
+        }
     };
 
     struct SwitchToTabArgs : public SwitchToTabArgsT<SwitchToTabArgs>
     {
         SwitchToTabArgs() = default;
         GETSET_PROPERTY(int32_t, TabIndex, 0);
+
+        static constexpr std::string_view TabIndexKey{ "index" };
+
+    public:
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // IMPORTANT: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<SwitchToTabArgs>();
+            if (auto tabIndex{ json[JsonKey(TabIndexKey)] })
+            {
+                args->_TabIndex = tabIndex.asInt();
+            }
+            return *args;
+        }
+    };
+
+    // Possible Direction values
+    static constexpr std::string_view LeftString{ "left" };
+    static constexpr std::string_view RightString{ "right" };
+    static constexpr std::string_view UpString{ "up" };
+    static constexpr std::string_view DownString{ "down" };
+
+    // Function Description:
+    // - Helper function for parsing a Direction from a string
+    // Arguments:
+    // - directionString: the string to attempt to parse
+    // Return Value:
+    // - The encoded Direction value, or Direction::Left if it was an invalid string
+    static TerminalApp::Direction ParseDirection(const std::string& directionString)
+    {
+        if (directionString == LeftString)
+        {
+            return TerminalApp::Direction::Left;
+        }
+        else if (directionString == RightString)
+        {
+            return TerminalApp::Direction::Right;
+        }
+        else if (directionString == UpString)
+        {
+            return TerminalApp::Direction::Up;
+        }
+        else if (directionString == DownString)
+        {
+            return TerminalApp::Direction::Down;
+        }
+        // default behavior for invalid data
+        return TerminalApp::Direction::Left;

```

Comment:
Uh, in what context is this usually used and is Left an appropriate default?

(That is... is it either sufficiently good for most circumstances such that a null is OK... or alternatively is it sufficiently bad that someone will instantly recognize they made a mistake?)